airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -368,20 +368,25 @@ def extract_xcom(self, pod: V1Pod) -> str:
                 _preload_content=False,
             )
         ) as resp:
-            result = self._exec_pod_command(resp, f'cat {PodDefaults.XCOM_MOUNT_PATH}/return.json')
+            result = self._exec_pod_command(resp, f'cat {PodDefaults.XCOM_MOUNT_PATH}/return.json', True)
             self._exec_pod_command(resp, 'kill -s SIGINT 1')
         if result is None:
             raise AirflowException(f'Failed to extract xcom from pod: {pod.metadata.name}')
         return result
 
-    def _exec_pod_command(self, resp, command: str) -> Optional[str]:
+    def _exec_pod_command(self, resp, command: str, extract_full_content=False) -> Optional[str]:
         if resp.is_open():
             self.log.info('Running command... %s\n', command)
             resp.write_stdin(command + '\n')
             while resp.is_open():
                 resp.update(timeout=1)
+                res = None
                 if resp.peek_stdout():
-                    return resp.read_stdout()
+                    res = resp.read_stdout()
+                    if extract_full_content:
```

Comment:
What is the value in keeping the previous broken behaviour? I'd say we just keep the while loop for all cases, I don't see why you'd want partial extraction especially if there is a risk of it throwing an exception when it is json loaded.

Reply:
_extract_xcom function can load content which is string and can be large also. The plus(+) operator works 
perfectly for string content and There is no surity about what all type values can be retuned by resp.read_stdout(). Now this function support two behavior one reading partial response and another reading full response. Keeping only the while loop may not be the right thing.

Reply:
I suppose we'd have to know what the original intention of this code was. But IMHO I don't think returning a partial response is a feature, it is a bug :)

Reply:
I have fixed review comments